bindings/transport-generic/src/main/java/esa/mo/mal/transport/gen/GENTransport.java

```diff
@@ -892,12 +916,13 @@ protected synchronized GENConcurrentMessageSender manageCommunicationChannel(
         } else {
             // outgoing message
             // get target URI
-            String remoteRootURI = getRootURI(msg.getHeader().getURITo().getValue());
+            String remoteRootURI = getRootURI(rerouteMessage(msg));
 
             // get sender if it exists
             sender = outgoingDataChannels.get(remoteRootURI);
 
-            if (null == sender) {
+            boolean firstTime = !connectionAttempts.contains(remoteRootURI);
+            if (null == sender && (connectWhenConsumerOffline || firstTime)) {
```

Comment:
The purpose of this property is to prevent the provider from trying to connect to the consumer when the connection is lost between them. If you want the provider to try to reconnect you can just set this property to `true` or don't set it at all because the default value is `true`.

Reply:
Ahh! Got it. Misread the `if`. Then it's clearer, but still a bit strange with a single connection attempt and never cleaning the list entry when reconnect property is set to false. What use case does it address?
Would it make sense to clear the list entry at some point?
Otherwise it's good as is, and let's keep the advanced log suppression for the future.

Reply:
The connection attempts set is used when the provider tries to connect to the supervisor. If the provider tries to connect to something (for example supervisor) and the property is set to `false` the `firstTime` check allows to create a message sender and connect to the target.

Reply:
I fixed the `firstTime` check. The `connectionAttempts` set is now updated regardless if the connection is successful or not.

I don't think clearing the `connectionAttempts` set makes sense because it would need to have tens of thousands of entries to use a meaningful amout of memory.